### PR TITLE
Multilabel support in ROC

### PR DESCRIPTION
**Pull request by SkafteNicki**
# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to update the docs?   
- [x] Did you write any new necessary tests?  

## What does this PR do?
Fixes https://github.com/PyTorchLightning/metrics/issues/103
Adds multilabel support to ROC metric

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by tchaton — APPROVED** [review]
LGTM ! Nice addition !

File: torchmetrics/classification/roc.py [hunk 1]
```diff
@@ -80,20 +85,42 @@ class ROC(Metric):
...
+        ...                      [0.8603, 0.0745, 0.1837]])
+        >>> target = torch.tensor([[1, 1, 0], [0, 1, 0], [0, 0, 0], [0, 1, 1]])
+        >>> roc = ROC(num_classes=3, pos_label=1)
+        >>> fpr, tpr, thresholds = roc(pred, target)
+        >>> fpr # doctest: +NORMALIZE_WHITESPACE
+        [tensor([0.0000, 0.3333, 0.3333, 0.6667, 1.0000]),
+         tensor([0., 0., 0., 1., 1.]),
+         tensor([0.0000, 0.0000, 0.3333, 0.6667, 1.0000])]
+        >>> tpr
+        [tensor([0., 0., 1., 1., 1.]), tensor([0.0000, 0.3333, 0.6667, 0.6667, 1.0000]), tensor([0., 1., 1., 1., 1.])]
```
Comment (Borda) [inline torchmetrics/classification/roc.py:103]:
```suggestion
        >>> tpr  # doctest: +NORMALIZE_WHITESPACE
        [tensor([0., 0., 1., 1., 1.]),
         tensor([0.0000, 0.3333, 0.6667, 0.6667, 1.0000]),
         tensor([0., 1., 1., 1., 1.])]
```

**Review by Borda — APPROVED** [review]

**Comment by Borda** [comment]
@SkafteNicki mind resolve conflict, then ready to land :]

**Review by justusschock — APPROVED** [review]

**Comment by Borda** [comment]
@SkafteNicki seems that the tests are not correct

**Comment by SkafteNicki** [comment]
@Borda thanks, had not run all test locally but they should be good now.
